tvbwidgets/exporters/model_exporters.py

```diff
@@ -0,0 +1,200 @@
+# -*- coding: utf-8 -*-
+#
+# "TheVirtualBrain - Widgets" package
+#
+# (c) 2022-2023, TVB Widgets Team
+#
+
+import abc
+import json
+import numpy
+import os
+import re
+from tvb.simulator.models import Model
+
+from tvbwidgets.core.exceptions import ModelExporterNotFoundError
+
+MODEL_CONFIGURATION_EXPORTS = {
+    'JSON': 'json',
+    'Python script': 'python'
+}
+
+
+def is_valid_file_name(filename):
+    # type: (str) -> bool
+    """
+    checks if a string is valid python file name
+    returns even if the file doesn't end with .py
+    """
+    patterns = [re.compile(r'\w+.py+$'), re.compile(r'\w+$')]
+    return any([re.match(p, filename) for p in patterns])
+
+
+class ABCModelExporter(abc.ABC):
+    @abc.abstractmethod
+    def __init__(self, model_instance, keys):
+        # type: (Model, list[str]) -> None
+        """
+        Abstract constructor to be implemented in subclasses
+        """
+        self.model_instance = model_instance
+        self.keys = keys
+        # name of the exported configuration is set
+        # from outside of class if we need other name than the default
+        self.config_name = self.default_config_name
+
+    @property
+    @abc.abstractmethod
+    def default_config_name(self):
+        # type: () -> str
+        """
+        Abstract property to be implemented in subclasses representing
+        default name of the configuration to be exported
+        """
+        pass
+
+    @abc.abstractmethod
+    def do_export(self):
+        """
+        Abstract method to be implemented in subclasses
+        implements export logic
+        """
+        pass
+
+    @staticmethod
+    def sanitize_property(prop: any) -> any:
+        # type: (any) -> any
```

Comment:
Shouldn't this belong to the JSONModelExporter?

Reply:
Since we might implement other exporters down the road which might make use of this method, I figured it would be useful to have this method here.

Reply:
I understand the argument, but it seems to me that this will be used only for JSON-like exporters, and if we were to implement more of those, they could extend the JSONModelExporter directly to obtain the same behavior. If we leave it here, PythonCodeExporter is polluted with a method that it does not need.